src/wifipaf/WiFiPAFEndPoint.cpp

```diff
@@ -797,81 +789,83 @@ SequenceNumber_t WiFiPAFEndPoint::AdjustRemoteReceiveWindow(SequenceNumber_t las
     return static_cast<uint8_t>(newRemoteWindowBoundary - newestUnackedSentSeqNum);
 }
 
-CHIP_ERROR WiFiPAFEndPoint::GetPktSn(Encoding::LittleEndian::Reader & reader, uint8_t * pHead, SequenceNumber_t & seqNum)
+CHIP_ERROR WiFiPAFEndPoint::GetPktSn(Encoding::LittleEndian::Reader reader, SequenceNumber_t & seqNum)
 {
     CHIP_ERROR err;
-    BitFlags<WiFiPAFTP::HeaderFlags> rx_flags;
-    size_t SnOffset = 0;
-    SequenceNumber_t * pSn;
-    err = reader.Read8(rx_flags.RawStorage()).StatusCode();
-    if (rx_flags.Has(WiFiPAFTP::HeaderFlags::kHankshake))
+    BitFlags<WiFiPAFTP::HeaderFlags> rxFlags;
+    size_t SkipOffset = 0;
+
+    err = reader.Read8(rxFlags.RawStorage()).StatusCode();
+    SuccessOrExit(err);
+    if (rxFlags.Has(WiFiPAFTP::HeaderFlags::kHandshake))
     {
         // Handkshake message => No ack/sn
         return CHIP_ERROR_INTERNAL;
     }
     // Always has header flag
-    SnOffset += kTransferProtocolHeaderFlagsSize;
-    if (rx_flags.Has(WiFiPAFTP::HeaderFlags::kManagementOpcode)) // Has Mgmt_Op
+    if (rxFlags.Has(WiFiPAFTP::HeaderFlags::kManagementOpcode))
     {
-        SnOffset += kTransferProtocolMgmtOpSize;
+        SkipOffset += kTransferProtocolMgmtOpSize;
     }
-    if (rx_flags.Has(WiFiPAFTP::HeaderFlags::kFragmentAck)) // Has ack
+    if (rxFlags.Has(WiFiPAFTP::HeaderFlags::kFragmentAck))
     {
-        SnOffset += kTransferProtocolAckSize;
+        SkipOffset += kTransferProtocolAckSize;
     }
-    pSn    = pHead + SnOffset;
-    seqNum = *pSn;
-
-    return CHIP_NO_ERROR;
+    err = static_cast<Encoding::LittleEndian::Reader &>(reader.Skip(SkipOffset)).Read8(&seqNum).StatusCode();
```

Comment:
Let's just do the Skip() on a separate line from the Read8 and avoid the casts?

And maybe file a followup issue to have Skip on the endianness-specific readers.

Reply:
Thank you for the suggestion. I will change it.